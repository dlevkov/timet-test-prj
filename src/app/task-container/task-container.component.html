<mat-card>
  <app-task-presenter *ngFor="let task of tasks$| async" [task]="task" (clicked)="onClick($event)"></app-task-presenter>
</mat-card>
