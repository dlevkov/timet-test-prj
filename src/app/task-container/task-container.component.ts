import { Component, OnInit } from '@angular/core';
import { Observable } from 'rxjs';
import { TaskModel } from '../models/task-model';
import { LogicService } from '../logic.service';

@Component({
  selector: 'app-task-container',
  templateUrl: './task-container.component.html',
  styleUrls: ['./task-container.component.scss'],
})
export class TaskContainerComponent implements OnInit {
  public tasks$: Observable<TaskModel[]>;
  constructor(private service: LogicService) {}

  ngOnInit() {
    this.tasks$ = this.service.tasks$;
  }

  public onClick(evt: TaskModel) {
    this.service.updateTask(evt);
  }
}
