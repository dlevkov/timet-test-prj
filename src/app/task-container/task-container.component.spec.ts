import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { TaskContainerComponent } from './task-container.component';
import { LogicService } from '../logic.service';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { MinuteSecondsPipe } from '../minute-seconds.pipe';

describe('TaskContainerComponent', () => {
  let component: TaskContainerComponent;
  let fixture: ComponentFixture<TaskContainerComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [TaskContainerComponent, MinuteSecondsPipe],
      providers: [{ provide: LogicService, useValue: jest.fn() }],
      schemas: [NO_ERRORS_SCHEMA],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TaskContainerComponent);
    component = fixture.componentInstance;
    // fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
