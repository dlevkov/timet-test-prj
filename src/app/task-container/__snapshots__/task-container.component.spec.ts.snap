// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskContainerComponent DOM should render tasks$ 1`] = `
<div
  id="root3"
  ng-version="7.2.11"
>
  <mat-card
    class="mat-card"
  >
    
    <app-task-presenter
      ng-reflect-task="[object Object],[object Object"
    >
      <p>
        [
  {
    "id": 1,
    "name": "test1",
    "buttonText": "pause",
    "timer": {
      "_isScalar": true,
      "value": 1
    }
  },
  {
    "id": 2,
    "name": "test2",
    "buttonText": "pause",
    "timer": {
      "_isScalar": true,
      "value": 1
    }
  },
  {
    "id": 3,
    "name": "test3",
    "buttonText": "pause",
    "timer": {
      "_isScalar": true,
      "value": 1
    }
  }
]
      </p>
    </app-task-presenter>
  </mat-card>
  <mat-card
    class="mat-card"
  >
    <mat-card-title
      class="mat-card-title"
    >
       Total: 00:01 
    </mat-card-title>
  </mat-card>
</div>
`;

exports[`TaskContainerComponent DOM should render totalTime$ 1`] = `
<div
  id="root2"
  ng-version="7.2.11"
>
  <mat-card
    class="mat-card"
  >
    
  </mat-card>
  <mat-card
    class="mat-card"
  >
    <mat-card-title
      class="mat-card-title"
    >
       Total: 00:35 
    </mat-card-title>
  </mat-card>
</div>
`;
